### PR TITLE
SATS: fix backwards compatibility

### PR DESCRIPTION
**Pull request by davydov-d**
## What
SATs declare widening of a data type is ok but in reality tests do not pass in a `'integer'-> '[integer, string]'` type change. Fixing it here

### CODE REVIEW

**Review by lazebnyi — APPROVED** [review]

**Review by sherifnada — CHANGES_REQUESTED** [review]
@davydov-d could you add a unit test to verify this behavior?

**Comment by davydov-d** [comment]
> @davydov-d could you add a unit test to verify this behavior?

@sherifnada sure, done. I've also separated spec validation from catalog validation because I believe the behavior should be different in these cases

**Review by alafanechere — CHANGES_REQUESTED** [review]
@davydov-d could you please explain your motivation for this change? Type widening sounds dangerous at first sight. Is it your approach to changing a spec type but still supporting the previous type?

**Comment by davydov-d** [comment]
> @davydov-d could you please explain your motivation for this change? Type widening sounds dangerous at first sight. Is it your approach to changing a spec type but still supporting the previous type?

@alafanechere Yes. [here](https://github.com/airbytehq/airbyte/pull/16400) I need to change data type from integer to string but support legacy integer values as well. That's why here I'm letting `<old_type> -> [<old_type>, <new_type>]` change to be valid (only for spec, not for catalog). Besides, it is stated to be a valid use case [here](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/utils/backward_compatibility.py#L79), but is not at the moment

File: airbyte-integrations/bases/source-acceptance-test/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,8 @@
 # Changelog
 
+## 0.2.3
+Backward compatibility tests: improve `check_if_type_of_type_field_changed` to make it less radical when validating specs and allow `'str' -> ['str', 'another_str']` type changes.[#16429](https://github.com/airbytehq/airbyte/pull/16429/)
```
Comment (alafanechere) [inline airbyte-integrations/bases/source-acceptance-test/CHANGELOG.md:4]: nit
```suggestion
Backward compatibility tests: improve `check_if_type_of_type_field_changed` to make it less radical when validating specs and allow `'str' -> ['str', '<another_type>']` type changes.[#16429](https://github.com/airbytehq/airbyte/pull/16429/)
```

**Review by alafanechere — APPROVED** [review]
Thank you for the clarification. LGTM

**Review by sherifnada — APPROVED** [review]

**Review by pedroslopez — APPROVED** [review]

**Comment by davydov-d** [comment]
/publish connector=bases/source-acceptance-test auto-bump-version=false

**Pull request merged**